mico-core/src/main/java/io/github/ust/mico/core/model/MicoService.java

```diff
@@ -146,20 +137,38 @@
         name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION,
         properties = {
             @ExtensionProperty(name = "title", value = "Service Crawling Origin"),
-            @ExtensionProperty(name = "x-order", value = "100"),
+            @ExtensionProperty(name = "x-order", value = "50"),
             @ExtensionProperty(name = "description", value = "Indicates where this service originates from.")
         }
     )})
-    private MicoServiceCrawlingOrigin serviceCrawlingOrigin;
+    @NotNull
```

Comment:
Why do you need to add **NotNull** if it is already initialized always with MicoServiceCrawlingOrigin.NOT_DEFINED?

Reply:
If we don't check if the property is `null`, the user of the API is able to set it to `null` explicitly. The default value is only used, if the property is not initialized.
The same applies to all array fields.

Reply:
I will consider to use the Spring configuration `spring.jackson.default-property-inclusion=NON_NULL` to ignore all `null` values. If this works, the `@NotNull` annotation is not needed anymore.

Reply:
My proposal with the Spring configuration does not work because it only configures the serialization of Jackson, not the deserialization.
Besides the `@NotNull` there is another option:
```
@JsonProperty("service-crawling-origin")
private MicoServiceCrawlingOrigin serviceCrawlingOrigin = MicoServiceCrawlingOrigin.NOT_DEFINED;

@JsonSetter("service-crawling-origin")
public MicoService setServiceCrawlingOrigin(MicoServiceCrawlingOrigin serviceCrawlingOrigin) {
    if (serviceCrawlingOrigin != null) {
        this.serviceCrawlingOrigin = serviceCrawlingOrigin;
    }
    return this;
}
```

This would be more user friendly because a `null` value does not lead to an exception. However it would be more difficult for the developer.

Reply:
I still would prefer the simple option with the `@NotNull` annotation.

Reply:
I also prefer the `@NotNull` annotation. Especially because we only have one user of our API: `mico-admin` (UI).

Reply:
@JanVetter What do you think?

Reply:
Finally I found a solution: `@JsonSetter(nulls = Nulls.AS_EMPTY)` and `@JsonSetter(nulls = Nulls.SKIP)`.
I will use `AS_EMPTY` for all arrays and strings that should be initialized to an empty string (`description`) and `SKIP` for all enum fields.